.github/workflows/deploy-to-azure-app-service.yml

```diff
@@ -0,0 +1,209 @@
+      build-verbosity:
+        required: false
+        type: string
+        default: quiet
+        description: Verbosity parameter for dotnet build.
+      build-enable-code-analysis:
+        required: false
+        type: string
+        default: "false"
+        description: Indicates whether to enable static code analysis during dotnet build.
+      build-enable-nuget-caching:
+        required: false
+        type: string
+        default: "false"
+        description: >
+          If set to "true", the NuGet packages will be cached by GitHub. This is faster if you have a lot of NuGet
+          packages, but negligible or even slower if you only have a handful.
+      build-enable-npm-caching:
+        required: false
+        type: string
+        default: "false"
+        description: >
+          If set to "true", the NPM packages will be cached by GitHub. This is faster if you have a lot of NPM packages,
+          but negligible or even slower if you only have a handful.
+      build-cache-version:
+        required: false
+        type: string
+        default: "0"
+        description: >
+          Change this to any other value to alter the cache key, effectively invalidating the latest cache. This is the
+          easiest current way to force "clear" the cache
+          (https://github.community/t/how-to-clear-cache-in-github-actions/129038/5) until the associated issue
+          (https://github.com/actions/cache/issues/2) is resolved and the feature is available not just via the RESTful
+          API.
+      timeout-minutes:
+        required: false
+        type: number
+        default: 360
+        description: Configuration for the timeout-minutes parameter of the workflow. The 360 is GitHub's default.
+      app-name:
+        required: true
+        type: string
+        description: What you see at the top of the blade on the Azure Portal. Can contain uppercase letters too.
+      slot-name:
+        required: true
+        type: string
+        description: What you see at the top of the blade on the Azure Portal, when you open the slot, before the app
+          name in parenthesis.
+      resource-group-name:
+        required: true
+        type: string
+        description: Name of the resource group.
+      url:
+        required: true
+        type: string
+        description: The URL of the web app slot.
+      web-project-path:
+        required: false
+        type: string
+        default: "./src/*Web/*.Web.csproj"
+        description: >
+          The path of the web project file to be processed. The value is given to PowerShell's `Get-ChildItem` cmdlet
+          as-is, so grepping is still possible but a name with spaces must be escaped separately.
+      runtime:
+        required: false
+        type: string
+        description: Publishes the application for a given runtime.
+      self-contained:
+        required: false
+        type: boolean
```

Comment:
While this would be nicer, use strings, just as for the other bool parameters for consistency. This is necessary because composite actions can only use string parameters. Later we might want to move the deployment to an action too, so this is also useful for future-proofing.